### ISSUE TITLE
KeyError running with --py36-plus on matplotlib's ticker.py

### ISSUE THREAD

**Issue by anntzer** [asottile/pyupgrade#52]
Running pyupgrade 1.5 on matplotlib's ticker.py as of, say, mpl3.0rc2 (https://github.com/matplotlib/matplotlib/blob/v3.0.0rc2/lib/matplotlib/ticker.py) gives me
```
Traceback (most recent call last):
  File "/usr/bin/pyupgrade", line 11, in <module>
    sys.exit(main())
  File "/usr/lib/python3.7/site-packages/pyupgrade.py", line 907, in main
    ret |= fix_file(filename, args)
  File "/usr/lib/python3.7/site-packages/pyupgrade.py", line 884, in fix_file
    contents_text = _fix_fstrings(contents_text)
  File "/usr/lib/python3.7/site-packages/pyupgrade.py", line 858, in _fix_fstrings
    tokens[i] = token._replace(src=_to_fstring(token.src, node))
  File "/usr/lib/python3.7/site-packages/pyupgrade.py", line 824, in _to_fstring
    name = ''.join((params[k or str(i)], dot, rest))
KeyError: '3'
```

**Comment by asottile** [comment]
Nice! thanks for the report, here's a minimal reproduction:

```python
a, b = 1, 2
print('{}{{}}{}'.format(a, b))
```

```console
$ python3 t.py
1{}2
$ pyupgrade --py36-plus t.py
Traceback (most recent call last):
  File "/tmp/pyupgrade/venv/bin/pyupgrade", line 11, in <module>
    load_entry_point('pyupgrade', 'console_scripts', 'pyupgrade')()
  File "/tmp/pyupgrade/pyupgrade.py", line 907, in main
    ret |= fix_file(filename, args)
  File "/tmp/pyupgrade/pyupgrade.py", line 884, in fix_file
    contents_text = _fix_fstrings(contents_text)
  File "/tmp/pyupgrade/pyupgrade.py", line 858, in _fix_fstrings
    tokens[i] = token._replace(src=_to_fstring(token.src, node))
  File "/tmp/pyupgrade/pyupgrade.py", line 824, in _to_fstring
    name = ''.join((params[k or str(i)], dot, rest))
KeyError: '3'
```

**Comment by asottile** [comment]
waiting for #54 and then I'll make a release -- thanks again for the report!

**Comment by asottile** [comment]
This has been released as part of v1.5.1 :tada:
